bin/spiced/src/lib.rs

```diff
@@ -108,12 +108,18 @@ pub async fn run(args: Args) -> Result<()> {
                 let data_source = Box::leak(data_source);
 
                 df.attach(ds, data_source, databackend::DataBackendType::default())
+                    .await
                     .context(UnableToAttachDataSourceSnafu {
                         data_source: source,
                     })?;
             }
             None => df
-                .attach_backend(&ds.name, databackend::DataBackendType::default())
+                .attach_backend(
```

Comment:
This breaks the scenario of creating an empty dataset that is used for ingesting data via DoPut (and in the future, the metrics endpoint)

Reply:
I don't think `attach_backend` is the correct abstraction here. A view is not a backend, its a view. It doesn't make sense for it to be a backend. Let's create a new method called `register_view` on the datafusion struct that we use here if `sql` is set.

Reply:
I agree from a customer standpoint and thought about this but ultimately I disagree from an implementation perspective. ViewTable is a TableProvider, it provides the backend implementation of the table. If it's not a backend here then neither is MemTable.

Reply:
MemTable is a backend because it handles ingesting data - a view doesn't

Reply:
That's fair